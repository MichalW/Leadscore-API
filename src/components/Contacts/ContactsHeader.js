import React from 'react';
import {Table} from 'semantic-ui-react';

const ContactsHeader = () => (
  <Table.Header>
    <Table.Row>
      <Table.HeaderCell>First name</Table.HeaderCell>
      <Table.HeaderCell>Last name</Table.HeaderCell>
      <Table.HeaderCell>Email</Table.HeaderCell>
    </Table.Row>
  </Table.Header>
);

export default ContactsHeader;
