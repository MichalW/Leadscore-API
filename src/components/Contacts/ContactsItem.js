import PropTypes from 'prop-types';
import React from 'react';
import get from 'lodash/get';
import {Table} from 'semantic-ui-react';

const ContactsItem = ({contact}) => (
  <Table.Row>
    <Table.Cell>{contact.firstName}</Table.Cell>
    <Table.Cell>{contact.lastName}</Table.Cell>
    <Table.Cell>{get(contact, ['emails', 0, 'email'])}</Table.Cell>
  </Table.Row>
);

ContactsItem.propTypes = {
  contact: PropTypes.shape({}).isRequired,
};

export default ContactsItem;
