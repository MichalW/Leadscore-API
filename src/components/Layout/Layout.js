import PropTypes from 'prop-types';
import React from 'react';
import {Container, Grid} from 'semantic-ui-react';

import Navbar from '../Navbar';
import Breadcrumb from '../Breadcrumb';

const Layout = ({children}) => (
  <Container>
    <Navbar />

    <Breadcrumb />

    <Grid centered>
      <Grid.Row>
        <Grid.Column>
          {children}
        </Grid.Column>
      </Grid.Row>
    </Grid>
  </Container>
);

Layout.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Layout;
