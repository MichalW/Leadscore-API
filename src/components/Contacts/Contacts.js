import PropTypes from 'prop-types';
import React, {Component} from 'react';
import {
  Header, Segment, Table, Icon,
} from 'semantic-ui-react';
import isEqual from 'lodash/isEqual';

import ContactsHeader from './ContactsHeader';
import ContactsItem from './ContactsItem';
import ContactsPagination from './ContactsPagination';
import Layout from '../Layout';
import {DEFAULT_CONTACTS_LIMIT} from '../../config';

class Contacts extends Component {
  componentDidMount() {
    this.contactsFetch();
  }

  componentDidUpdate({match: {params: prevParams}}) {
    const {match: {params}} = this.props;

    if (!isEqual(params, prevParams)) {
      this.contactsFetch();
    }
  }

  contactsFetch = () => {
    const {contactsFetchRequest, match: {params}} = this.props;

    const offset = DEFAULT_CONTACTS_LIMIT * (params.page - 1);

    contactsFetchRequest(offset, DEFAULT_CONTACTS_LIMIT);
  };

  render() {
    const {contactsData} = this.props;

    return (
      <Layout>
        <Segment>
          <Header as="h2">
            <Icon name="users" />
            <Header.Content>
              Contacts
              <Header.Subheader>Manage your contacts</Header.Subheader>
            </Header.Content>
          </Header>

          <Table celled selectable>
            <ContactsHeader />

            <Table.Body>
              {contactsData.map(contact => (
                <ContactsItem contact={contact} key={contact.id} />
              ))}
            </Table.Body>
          </Table>

          <ContactsPagination />
        </Segment>
      </Layout>
    );
  }
}


Contacts.propTypes = {
  contactsData: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  contactsFetchRequest: PropTypes.func.isRequired,
  match: PropTypes.shape({}).isRequired,
};

export default Contacts;
