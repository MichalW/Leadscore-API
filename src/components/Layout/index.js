import {connect} from 'react-redux';

import Layout from './Layout';

const mapStateToProps = state => ({
  signedIn: state.authentication.signedIn,
  username: state.authentication.user.username,
});

export default connect(mapStateToProps)(Layout);
