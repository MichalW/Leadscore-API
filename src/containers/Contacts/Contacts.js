import React from 'react';
import {Segment} from 'semantic-ui-react';

import Header from '../../components/Header';
import Layout from '../../components/Layout';
import ContactsList from '../../components/ContactsList';

const Contacts = () => (
  <Layout>
    <Segment>
      <Header title="Contacts" icon="users" subTitle="Manage your contacts" />

      <ContactsList />
    </Segment>
  </Layout>
);

export default Contacts;
