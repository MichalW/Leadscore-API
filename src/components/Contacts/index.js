import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';

import Contacts from './Contacts';
import {contactsFetchRequest} from '../../actions/contacts';

const mapStateToProps = state => ({
  contactsData: state.contacts.data,
  contactsCount: state.contacts.count,
});

const mapDispatchToProps = dispatch => ({
  contactsFetchRequest: bindActionCreators(contactsFetchRequest, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(Contacts);
