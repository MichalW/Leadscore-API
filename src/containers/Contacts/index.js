import {connect} from 'react-redux';

import Contacts from './Contacts';

const mapStateToProps = state => ({
  username: state.authentication.user.username,
});

export default connect(mapStateToProps)(Contacts);
