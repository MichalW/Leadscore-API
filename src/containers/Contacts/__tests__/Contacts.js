import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';

import Contacts from '../Contacts';
import Layout from '../../../components/Layout';

it('should render Contacts', () => {
  const props = {
    username: 'ExampleUsername',
  };

  const contacts = shallow(<Contacts {...props} />);

  expect(contacts.find(Layout).exists()).toBe(true);
  expect(toJson(contacts)).toMatchSnapshot();
});
