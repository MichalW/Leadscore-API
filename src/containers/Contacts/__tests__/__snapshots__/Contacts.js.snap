// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Home 1`] = `
<Layout>
  <Segment>
    <HeaderComponent
      icon="users"
      subTitle="Manage your contacts"
      title="Contacts"
    />
    <withRouter(Connect(ContactsList)) />
  </Segment>
</Layout>
`;
